op-chain-ops/interopgen/recipe.go

```diff
@@ -0,0 +1,280 @@
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+)
+
+type InteropDevRecipe struct {
+	L1ChainID        uint64
+	L2ChainIDs       []uint64
+	GenesisTimestamp uint64
+}
+
+func (r *InteropDevRecipe) Build(addrs devkeys.Addresses) (*WorldConfig, error) {
+	// L1 genesis
+	l1Cfg := &L1Config{
+		ChainID: new(big.Int).SetUint64(r.L1ChainID),
+		DevL1DeployConfig: genesis.DevL1DeployConfig{
+			L1BlockTime:             6,
+			L1GenesisBlockTimestamp: hexutil.Uint64(r.GenesisTimestamp),
+			L1GenesisBlockGasLimit:  30_000_000,
+		},
+		Prefund: make(map[common.Address]*big.Int),
+	}
+
+	l1Users := devkeys.ChainUserKeys(l1Cfg.ChainID)
+	for i := uint64(0); i < 20; i++ {
+		userAddr, err := addrs.Address(l1Users(i))
+		if err != nil {
+			return nil, fmt.Errorf("failed to get L1 user addr %d: %w", i, err)
+		}
+		l1Cfg.Prefund[userAddr] = Ether(10_000_000)
+	}
+
+	superchainOps := devkeys.SuperchainOperatorKeys(l1Cfg.ChainID)
+
+	superchainDeployer, err := addrs.Address(superchainOps(devkeys.SuperchainDeployerKey))
+	if err != nil {
+		return nil, err
+	}
+	superchainProxyAdmin, err := addrs.Address(superchainOps(devkeys.SuperchainProxyAdminOwner))
+	if err != nil {
+		return nil, err
+	}
+	superchainProtocolVersionsOwner, err := addrs.Address(superchainOps(devkeys.SuperchainProtocolVersionsOwner))
+	if err != nil {
+		return nil, err
+	}
+	superchainConfigGuardian, err := addrs.Address(superchainOps(devkeys.SuperchainConfigGuardianKey))
+	if err != nil {
+		return nil, err
+	}
+	l1Cfg.Prefund[superchainDeployer] = Ether(10_000_000)
+	l1Cfg.Prefund[superchainProxyAdmin] = Ether(10_000_000)
+	l1Cfg.Prefund[superchainConfigGuardian] = Ether(10_000_000)
+	superchainCfg := &SuperchainConfig{
+		ProxyAdminOwner:       superchainProxyAdmin,
+		ProtocolVersionsOwner: superchainProtocolVersionsOwner,
+		Deployer:              superchainDeployer,
+		Implementations: OPSMImplementationsConfig{
+			Release: "op-contracts/0.0.1",
+			FaultProof: SuperFaultProofConfig{
+				WithdrawalDelaySeconds:          big.NewInt(604800),
+				MinProposalSizeBytes:            big.NewInt(10000),
+				ChallengePeriodSeconds:          big.NewInt(120),
+				ProofMaturityDelaySeconds:       big.NewInt(12),
+				DisputeGameFinalityDelaySeconds: big.NewInt(6),
+				// Legacy config:
+				//UseFaultProofs:                  true,
+				//FaultGameAbsolutePrestate:       common.HexToHash("0x03c7ae758795765c6664a5d39bf63841c71ff191e9189522bad8ebff5d4eca98"),
+				//FaultGameMaxDepth:               50,
+				//FaultGameClockExtension:         0,
+				//FaultGameMaxClockDuration:       1200,
+				//FaultGameGenesisBlock:           0,
+				//FaultGameGenesisOutputRoot:      common.HexToHash("0xDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEF"),
+				//FaultGameSplitDepth:             14,
+				//RespectedGameType:               254, // "fast" game type
```

Comment:
Why not remove this?

Reply:
Oh these were my notes while migrating from the old deploy system to the new OPSM one. They can be removed now. Will clean it up.